internal/logger/logger.go

```diff
@@ -15,34 +15,60 @@
 package logger
 
 import (
+	"fmt"
 	"io"
 	"log"
+	"log/syslog"
 	"os"
 
 	"github.com/jacobsa/daemonize"
 )
 
+// ProgrammeName constant is used while writing the logs to syslog file, it is
+// used while filtering the gcsfuse log-message from the syslog file, since
+// syslog file contains all the system related logs from other programmes too.
+const ProgrammeName string = "gcsfuse"
+
 var (
 	defaultLoggerFactory *loggerFactory
 	defaultInfoLogger    *log.Logger
 )
 
 // InitLogFile initializes the logger factory to create loggers that print to
 // a log file.
+// In case of empty file, it starts writing the log to syslog file, which
+// is eventually filtered and redirected to a fixed location using syslog
+// config.
 func InitLogFile(filename string, format string) error {
-	f, err := os.OpenFile(
-		filename,
-		os.O_WRONLY|os.O_CREATE|os.O_APPEND,
-		0644,
-	)
-	if err != nil {
-		return err
+	var f *os.File
+	var sysWriter *syslog.Writer
+	var err error
+	if filename != "" {
+		f, err = os.OpenFile(
+			filename,
+			os.O_WRONLY|os.O_CREATE|os.O_APPEND,
+			0644,
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Priority consist of facility and severity, here facility to specify the
+		// type of system that is logging the message to syslog and severity is log-level.
+		// Here, we are using facility as syslog.LOG_LOCAL7 - is a custom facility syslog
```

Comment:
instead method user applications are allowed to use any value between local0 to local7 and we are using local7.